src/overlays/actors/ovl_En_Zl1/z_en_zl1.c

```diff
@@ -15,51 +15,650 @@ void EnZl1_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    Vec3f sp48;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80B4B874(thisx, globalCtx);
+    if (globalCtx->csCtx.state == 0) {
+        this->actionFunc = func_80B4BBC4;
+        return;
+    }
+
+    actorAction = globalCtx->csCtx.npcActions[0];
+    if (actorAction != NULL) {
+        func_80B4B7F4(actorAction, &sp74);
+        func_80B4B834(actorAction, &sp68);
+        if (this->unk_1E6 == 0) {
+            sp48 = sp74;
+            thisx->initPosRot.pos = sp48;
+            thisx->posRot.pos = sp48;
+        }
+        if (this->unk_1E6 != actorAction->action) {
+            frameCount = SkelAnime_GetFrameCount(&spB0.unk_00[actorAction->action]->genericHeader);
+            SkelAnime_ChangeAnim(&this->skelAnime, spB0.unk_00[actorAction->action], 1.0f, 0.0f, frameCount,
+                                 spA4.unk_00[actorAction->action], -10.0f);
+            this->unk_1E6 = actorAction->action;
+        }
+        thisx->velocity = velocity;
+        if (globalCtx->csCtx.frames < actorAction->endFrame) {
+            actionLength = actorAction->endFrame - actorAction->startFrame;
+            thisx->velocity.x = (sp68.x - sp74.x) / actionLength;
+            thisx->velocity.y = (sp68.y - sp74.y) / actionLength;
+            thisx->velocity.y += thisx->gravity;
+            if (thisx->velocity.y < thisx->minVelocityY) {
+                thisx->velocity.y = thisx->minVelocityY;
+            }
+            thisx->velocity.z = (sp68.z - sp74.z) / actionLength;
+        }
+        func_80038290(globalCtx, thisx, &this->unk_200, &this->unk_206, thisx->posRot2.pos);
+        func_800C04D8(globalCtx, this->unk_1E8, &sp98, &sp8C);
+        func_800C0704(globalCtx, this->unk_1E8, 70.0f);
+    }
+}
+
+void func_80B4BBC4(Actor* thisx, GlobalContext* globalCtx) {
+    EnZl1* this = THIS;
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000438.genericHeader);
+    Player* player = PLAYER;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000438, 1.0f, 0.0f, frameCount, 0, 0.0f);
+    func_8002DF54(globalCtx, thisx, 1);
+    func_8002F7DC(&player->actor, 0x6836);
+    thisx->textId = 0x7039;
+    func_8010B680(globalCtx, thisx->textId, 0);
+    this->unk_1E2 = 0;
+    this->actionFunc = func_80B4BF2C;
+}
+
+void func_80B4BC78(Actor* thisx, GlobalContext* globalCtx) {
+    AH_A sp90 = D_80B4E6D4;
+    u8_A sp84 = D_80B4E6FC;
+    EnZl1* this = THIS;
+    f32 actionLength;
+    Vec3f sp70;
+    Vec3f sp64;
+    Vec3f velocity = D_80B4E708;
+    CsCmdActorAction* actorAction;
+    s32 pad;
+    f32 frameCount;
+    Vec3f sp40;
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->skelAnime.animCurrentSeg == &D_06010B38) {
+            frameCount = SkelAnime_GetFrameCount(&D_06011348.genericHeader);
+            SkelAnime_ChangeAnim(&this->skelAnime, &D_06011348, 1.0f, 0.0f, frameCount, 0, -10.0f);
+        }
+    }
+    func_80B4B874(this, globalCtx);
+    actorAction = globalCtx->csCtx.npcActions[0];
```

Comment:
actor action was recently renamed to npcActions so I would update that in all of your variable names in this file